src/braket/circuits/circuit.py

```diff
@@ -773,6 +822,40 @@ def apply_initialization_noise(
 
         return apply_noise_to_moments(self, noise, target_qubits, "initialization")
 
+    def set_parameter_values(self, param_values: Dict[str, Number]):
+        """
+        Sets FreeParameters based upon their name and values passed in.
+
+        Args:
+            param_values Dict[str, Number]:  A mapping of FreeParameter names
+                to a value to assign to them.
+
+        Raises:
+            ValueError: If there are no parameters that match the key for the arg
+                param_values.
+        """
+        for param in param_values:
+            param_valid = False
+            for free_param in self.parameters:
+                if param == free_param.name:
+                    param_valid = True
+                    free_param.fix_value(param_values[param])
+            if not param_valid:
+                raise ValueError(f"No parameter in the circuit named: {param}")
+
+    def get_parameter_values(self) -> Dict[FreeParameter, Number]:
+        """
+        Populates a dictionary of the FreeParameters to their values in this circuit.
+
+        Returns:
+            Dict[FreeParameter, Number]: Returns a mapping of FreeParameter and
+                the corresponding value.
+        """
+        param_values = {}
+        for param in self.parameters:
+            param_values[param] = param.parameter_value
```

Comment:
What happens if two parms (with the same name) have two different values? Do we ever check for that?

Reply:
Removed function.